src/snowflake/snowpark/session.py

```diff
@@ -191,6 +191,38 @@ class Session:
     A :class:`Session` object is not thread-safe.
     """
 
+    class RuntimeConfig:
+        def __init__(self, session: "Session", conf: Dict[str, Any]) -> None:
+            self._session = session
+            for key, val in conf.items():
+                if hasattr(Session, key):
+                    setattr(session, key, val)
+
+        def get(self, key: str, default=None) -> Any:
+            if hasattr(Session, key):
+                return getattr(self._session, key)
+            if hasattr(self._session._conn._conn, key):
+                return getattr(self._session._conn._conn, key)
+            return default
+
+        def is_modifiable(self, key: str) -> bool:
+            if hasattr(Session, key) and isinstance(getattr(Session, key), property):
+                return getattr(Session, key).fset is not None
+            if hasattr(SnowflakeConnection, key) and isinstance(
+                getattr(SnowflakeConnection, key), property
+            ):
+                return getattr(SnowflakeConnection, key).fset is not None
```

Comment:
I'm curious: if `fset` is `None`, what happens if we try to do `setattr`? I guess it still works and that's why we need a check first?

Reply:
`fset` is None means the `property` has no setter, this is for properties that `host` or `port` that is not meant to be changed. `setattr` on a `property` with no setter should generate `AttributeError: can't set attribute '<name>'`. In this way, we could distinguish mutable and immutable configs.

Reply:
If `setattr` throws an error for immutable configs, can we skip this check and directly do `setattr` and let it raise `AttributeError` for the immutable case?

Reply:
We could, but we still need to check for whether `key` is an existing property to make sure users can't abuse `setattr`. Is there a specific reason you want this behavior?

Reply:
I see. I'm just trying to see if there is a possibility to reduce code complexity, because this mutability check is quite complicated :)